rt/rt.mjs

```diff
@@ -200,12 +224,20 @@ export class Engine {
         } else if (tag == TAGS.string) {
             let x = ptr;
             let s = "";
-            while (x != CONSTANTS[null]) {
-                const c = this.jsFromScheme(this.carOf(x));
-                s += c;
-                x = this.cdrOf(x);
+            if (extract_tag(this.carOf(x)) == TAGS.fixnum) {
+                // this is a new style string.
+                const length = this.jsFromScheme(this.carOf(x));
```

Comment:
I think this is from the string PR but it looks good.  Might as well include a commit changing to remove the old-string case.